### PR TITLE
csv and excel file read and parse into json

### PR DESCRIPTION
**Pull request by Radheshhathwar**

### CODE REVIEW

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Autowired
+    private CsvDataImporterService csvDataImporterService;
+
+    String csvFilePath = "/home/radhesh/Downloads/artifact_bug_igot-kb.csv";
+    String excelFilePath = "/home/radhesh/Desktop/examsAndAdmissions/src/main/resources/Template_Examcycle_Creation.xlsx";
+
+    JSONArray jsonArray = null;
+
+    @PostMapping("/fileTest")
+    public ResponseEntity<Map<String, Object>> handleFileUploads(@RequestParam("file") MultipartFile file, @RequestParam("fileType") String fileType) {
```
Comment (shishirsuman092) [inline src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java]: What is use of csvFilePath?
if it is to read file from the location then why are you expecting client to send multipart file in request?

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        Map<String, Object> response = new HashMap<>();
+
+        try {
+            if ("csv".equalsIgnoreCase(fileType)) {
```
Comment (shishirsuman092) [inline src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java]: put CSV and EXCEL in constants

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                // Call the CSV processing method
+                jsonArray = csvDataImporterService.csvToJson(csvFilePath);
+
+                response.put("data", jsonArray.toString());
+                return ResponseEntity.ok(response);
+            } else if ("excel".equalsIgnoreCase(fileType)) {
+                // Call the Excel processing method
+                jsonArray = csvDataImporterService.excelToJson(excelFilePath);
+
+                response.put("data", jsonArray.toString());
```
Comment (shishirsuman092) [inline src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java]: Write SWITCH statement if there are multiple if else

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/service/CsvDataImporterService.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        for (Map.Entry<String, Integer> entry : headerMap.entrySet()) {
+            String header = entry.getKey();
+            int columnIndex = entry.getValue();
+            String value = csvRecord.get(columnIndex);
+            try {
+                jsonObject.put(header, value);
+            } catch (JSONException e) {
+                throw new RuntimeException(e);
+            }
+        }
```
Comment (shishirsuman092) [inline src/main/java/com/tarento/upsmf/examsAndAdmissions/service/CsvDataImporterService.java]: 1. use stream for better readability
2. why aren't you using FasterXMLJackson as it is provided by spring. use below code.

> @Autowired
private ObjectMapper objectMapper;

**Review by shishirsuman092 — CHANGES_REQUESTED** [review]

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java (hunk 1 above)
Comment (Radheshhathwar) [reply src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java]: Removed the additional path variables and using the multipartfile variable

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java (hunk 2 above)
Comment (Radheshhathwar) [reply src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java]: Done added them

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java (hunk 3 above)
Comment (Radheshhathwar) [reply src/main/java/com/tarento/upsmf/examsAndAdmissions/controller/FileUploadController.java]: used switch case

File: src/main/java/com/tarento/upsmf/examsAndAdmissions/service/CsvDataImporterService.java (hunk 4 above)
Comment (Radheshhathwar) [reply src/main/java/com/tarento/upsmf/examsAndAdmissions/service/CsvDataImporterService.java]: Made the changes

**Review by shishirsuman092 — APPROVED** [review]

**Pull request merged**